jwst/extract_1d/extract.py

```diff
@@ -2862,20 +2869,18 @@ def do_extract1d(input_model, ref_dict, smoothing_length=None,
                         "so apply_nod_offset will be set to False",
                         source_type)
 
-        if isinstance(input_model, (datamodels.ImageModel,
-                                    datamodels.DrizProductModel)):
+        if isinstance(input_model, datamodels.ImageModel):
```

Comment:
There's one failing regtest (test_nirspec_fs_spec3) that shows we need a slight change in this line. The failing regtest is an instance of NIRSpec fixed-slit data for a single slit, which I think used to get resampled into a `DrizProductModel` (because it's only 1 slit). Initially this PR changed that result to an `ImageModel`, but now it's been changed again such that a resampled NIRSpec single slit gets stored in `SlitModel`. Hence it's no longer getting into this branch that sets the proper `slitname` value for it and hence later on it fails when trying to find a matching slit name ("apername") in the extract1d reference file. It's slitname is set to "NRS_FIXEDSLIT", but it needs to be "S400A1" in this case. So if you change this line to test for both `ImageModel` and `SlitModel`, it then works properly, e.g.
```
  if isinstance(input_model, (datamodels.ImageModel, datamodels.SlitModel)):
```

Reply:
That won't work.  NRS_BRIGHTOBJ data are SlitModel, and they can't be lumped into this case.  Brightobj data (currently) need slitname = "NRS_BRIGHTOBJ" in order to select the correct entry in the reference file.  I would much rather this type of data had the correct slit name, though.

Reply:
Though maybe ImageModel and 3-D SlitModel should be lumped into one case, with the number if iterations set to 1 for ImageModel.

Reply:
Hmmm, good point about NRS_BRIGHTOBJ. Hadn't thought of that case. There's a dedicated extract1d ref file for NRS_BRIGHTOBJ that contains entries for both "S1600A1" and "NRS_BRIGHTOBJ". Since it's already a dedicated ref file for the bright object mode, I would think we could go down to just 1 entry in that ref file and have it use the proper aperture name, which is "S1600A1". The param values could be set to whatever's appropriate for when that slit is used for BRIGHTOBJ, and then meanwhile the "S1600A1" entry in the regular extract1d ref file would have params appropriate for non-TSO mode. Then the proposed logic change here should still be OK.

But what about NIS_SOSS? Does that also use `SlitModel`, with 3-D arrays, or does it use `CubeModel`?

Reply:
The difference between those two entries (one for S1600A1 and the other for NRS_BRIGHTOBJ) is that the former has `extract_width` 7 and the latter has `extract_width` 32.

Plain old NIRISS SOSS is a 2-D ImageModel.  I don't know what data model is used for the corresponding TSO mode.

Reply:
could we do something like (I know if it not elegant) 
if input_model.meta.exposure.type != 'NRS_BRIGHTOBJ' and  isinstance(input_model, (datamodels.ImageModel, datamodels.SlitModel)):
and then also keep 
 elif isinstance(input_model, (datamodels.CubeModel,
                                      datamodels.SlitModel)):  (this should catch NRS_BRIGHTOBJ case)

Reply:
Another solution that appears to work is to copy the existing the if-block
```
if input_model.meta.exposure.type == 'NRS_FIXEDSLIT':
    slitname = input_model.meta.instrument.fixed_slit
```
down in the `elif` block that includes `CubeModel` and `SlitModel`. That way it's not necessary to add `SlitModel` to the branch that handles `ImageModel` and it leaves "NRS_BRIGHTOBJ" (and NIS_SOSS) data alone.

But if the two of you can come up with a more elegant way of combining these two branches, then none of this should be needed.

Reply:
Yes, I think it would work to move the section for using `fixed_slit` for `slitname` for NRS_FIXEDSLIT data to the block for CubeModel and SlitModel, and I would add ImageModel to that block (i.e. so it would be for ImageModel, CubeModel, and SlitModel).  Then it should be OK to delete the block that's currently for ImageModel.  The block for CubeModel, etc., already has a test for `len(shape) == 2`.  This might take some tinkering to make sure `slitname` gets set appropriately for all data of these types.
I would also suggest uncommenting the lines for setting `slitname = input_model.name` (which would be for NRS_BRIGHTOBJ), if possible.  And I agree with @hbushouse regarding replacing the reference file for brightobj data to have just one entry, for slit "S1600A1".